src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -863,6 +869,10 @@ private void visitTypeAsPrimary(JSType type) {
     }
 
     private void visitType(JSType typeToVisit) {
+      visitType(typeToVisit, false);
```

Comment:
Nit: maybe add an inline `/* extendingInstanceClass */` for the parameter name in call sites? It's a bit hard to read like this.